src/pages/MovieSelected.js

```diff
@@ -1,31 +1,64 @@
-import React, {Component} from 'react';
-import Footer from '../components/footer/Footer';
+import React, { Component } from 'react';
+import connect from 'react-redux/es/connect/connect';
 import SearchResults from '../components/search-result/SearchResults';
 import SelectedFilm from '../components/navbar/seleced-film/SelectedFilm';
-import Navbar from "../components/navbar/Navbar";
-import SortBarInfo from "../components/sort-bar/sort-bar-info/SortBarInfo";
-import SortBar from "../components/sort-bar/SortBar";
-import {MovieSearchService} from "../serveses/move-search/MovieSearchService";
-import MovieSearch from "./MovieSearch";
+import Navbar from '../components/navbar/Navbar';
+import SortBarInfo from '../components/sort-bar/sort-bar-info/SortBarInfo';
+import SortBar from '../components/sort-bar/SortBar';
+import MovieSearch from './MovieSearch';
+import {
+  getFilmRequest,
+  searchFilmRequest,
+} from '../store/movie-search/actions';
 
 class MovieSelected extends Component {
+  componentDidMount() {
+    const { id, searchValue, searchBy } = this.props;
+    this.props.getFilmRequest(id);
+    this.props.searchFilmRequest(searchValue, searchBy);
+  }
+
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    const { id, searchValue, searchBy } = this.props;
+    const oldId = prevProps.id;
+    if (oldId !== id) {
+      this.props.getFilmRequest(id);
+      this.props.searchFilmRequest(searchValue, searchBy);
+    }
+  }
+
+
   render() {
-    const film = MovieSearchService.getFilmById(this.props.match ? this.props.match.params.id : "");
-    if (!film) {
-      return <MovieSearch/>
+    const { film, films } = this.props;
+    if (!film || !film.genres) {
+      return <MovieSearch />;
     }
-    let films = MovieSearchService.filterFilms('genre', film.genre).filter(value => value.id !== film.id);
-    let sortBarInfo = "Films by " + film.genre + " genre";
+    const sortBarInfo = `Films by ${film.genres[0]} genre`;
 
     return (
       <div>
-        <Navbar children={<SelectedFilm film={film}/>}/>
-        <SortBar info={<SortBarInfo text={sortBarInfo}/>}/>
-        <SearchResults films={films}/>
-        <Footer/>
+        <Navbar>
+          <SelectedFilm film={film} />
+        </Navbar>
+        <SortBar>
+          <SortBarInfo text={sortBarInfo} />
+        </SortBar>
+        <SearchResults films={films} />
```

Comment:
I think that films should be requested in `SearchResults` component to make it more reusable. It should be a smart component.